test/unit/fiber_channel.cc

```diff
@@ -78,12 +78,167 @@ fiber_channel_get()
 	status = check_plan();
 }
 
+static void
+fiber_channel_close_basic(enum fiber_channel_close_mode mode)
+{
+	fiber_channel_set_close_mode(mode);
+
+	bool graceful = mode == FIBER_CHANNEL_CLOSE_GRACEFUL;
+	struct fiber_channel *channel = fiber_channel_new(10);
+
+	char msg_1;
+	char msg_2;
+	char msg_3;
+	void *ptr = NULL;
+
+	ok(fiber_channel_put_timeout(channel, &msg_1, 0) == 0,
+	   "fiber_channel_put(msg_1)");
+
+	ok(fiber_channel_put_timeout(channel, &msg_2, 0) == 0,
+	   "fiber_channel_put(msg_2)");
+
+	fiber_channel_get(channel, &ptr);
+
+	ok(ptr == &msg_1, "fiber_channel_get(1)");
+
+	fiber_channel_close(channel);
+
+	ok(channel->is_closed, "is_closed");
+	ok(channel->is_destroyed == !graceful,
+	   graceful ? "not is_destroyed" : "is_destroyed");
+
+	ok(fiber_channel_put_timeout(channel, &msg_3, 0) != 0,
+	   "not fiber_channel_put(msg_3)");
+
+	ptr = NULL;
+	fiber_channel_get(channel, &ptr);
+
+	ok(ptr == (graceful ? &msg_2 : NULL),
+	   graceful ? "fiber_channel_get(1)" : "not fiber_channel_get(1)");
+	ok(channel->is_destroyed, "is_destroyed");
+
+	fiber_channel_delete(channel);
+}
+
+static int
+reader_f(va_list ap)
+{
+	struct fiber_channel *channel = (struct fiber_channel *)fiber()->f_arg;
```

Comment:
Nit: An explicit casting `void *` to a specific pointer type is not necessary. Same as with `struct foo *x = malloc(sizeof(struct foo));`

Reply:
Removed this in both cases.

Reply:
Returned back, since implicit conversion from `void *` is not allowed in C++.

Reply:
Sorry. I missed that it is a `.cc` file.